wire-ios/Wire-iOS/Sources/SwiftUI Helpers/HapticFeedbackModifier.swift

```diff
@@ -0,0 +1,50 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+
+func withFeedback(
+    _ style: UINotificationFeedbackGenerator.FeedbackType,
+    _ action: @escaping () -> Void
+) -> () -> Void {
+    { () in
+        let feedback = UINotificationFeedbackGenerator()
+        feedback.prepare()
+        feedback.notificationOccurred(style)
+        action()
+    }
+}
```

Comment:
Why is this a free function?

Reply:
"Free" as it's not part of any extension?

Reply:
yes, it's define at the top level and isn't attached to any type.